ipwb/replay.py

```diff
@@ -255,13 +255,33 @@ def show_uri(path, datetime=None):
 
     for idx, hLine in enumerate(hLines):
         k, v = hLine.split(': ', 1)
+
+        if k.lower() == 'transfer-encoding' and v.lower() == 'chunked':
+            resp.set_data(extractResponseFromChunkedData(payload))
         if k.lower() != "content-type":
             k = "X-Archive-Orig-" + k
+
         resp.headers[k] = v
 
     return resp
 
 
+def extractResponseFromChunkedData(data):
+    chunkDescriptor = -1
+    retStr = ''
+
+    (chunkDescriptor, rest) = data.split('\n', 1)
+    while chunkDescriptor.strip() != '0':
+        chunkDecFromHex = int(chunkDescriptor, 16)  # Get dec for slice
```

Comment:
The `chunkDescriptor` line is not always just a purely hexadecimal number. If there are some extensions used, it may have some more stuff after a semicolon. If we are only interested in the number of octets for now then at least the `chunkDescriptor` should be split on `;` and the first value should be passed to the `int()` typecasting function or else be ready to face a `ValueError`. By extracting chunk size from chunk descriptor line, we can then set the termination condition of the while loop to compare with integer 0 instead of string 0.

[From Wikipedia:](https://en.wikipedia.org/wiki/Chunked_transfer_encoding#Format)

> Each chunk starts with the number of octets of the data it embeds expressed as a hexadecimal number in ASCII followed by optional parameters (chunk extension) and a terminating CRLF sequence, followed by the chunk data. The chunk is terminated by CRLF.

> If chunk extensions are provided, the chunk size is terminated by a semicolon and followed by the parameters, each also delimited by semicolons. Each parameter is encoded as an extension name followed by an optional equal sign and value. These parameters could be used for a running message digest or digital signature, or to indicate an estimated transfer progress, for instance.